model-providers/openai/openai-common/runtime/src/main/java/io/quarkiverse/langchain4j/openai/OpenAiRestApi.java

```diff
@@ -182,35 +183,25 @@ public boolean test(SseEvent<String> event) {
         }
     }
 
-    interface AuthProvider {
-        String getAuthorization(Input input);
-
-        interface Input {
-            String method();
-
-            URI uri();
-
-            MultivaluedMap<String, Object> headers();
-        }
-    }
-
     class OpenAIRestAPIFilter implements ResteasyReactiveClientRequestFilter {
-        AuthProvider authorizer;
+        ModelAuthProvider authorizer;
 
-        public OpenAIRestAPIFilter(AuthProvider authorizer) {
+        public OpenAIRestAPIFilter(ModelAuthProvider authorizer) {
```

Comment:
IMHO it should be aligned with how it is done for Vertex AI Gemini `TokenFilter`, to have `ExecutorService` injected as well and then call a custom provider from this `ExecutorService` as providers may want to use some blocking calls etc. Using `ManagedExecutor` is best as it retains the original `RequestScope` which lets users inject Quarkus API beans which require it, as proposed in #708.

Reply:
Good point!